homeassistant/components/recorder.py

```diff
@@ -194,6 +204,7 @@ def run(self):
         """Start processing events to save."""
         self._setup_connection()
         self._setup_run()
+        self._purge_old_data()
```

Comment:
I think it would make sense to schedule this some time after launch so it won't slow down the launch of Home Assistant.

```python
from homeassistant.helpers.event import track_point_in_utc_time

track_point_in_utc_time(self.hass, lambda now: self._purge_old_data(),
                        dt_util.utcnow() + timedelta(minutes=5)
```